Gems/EMotionFX/Code/EMotionFX/Source/BlendTreeFabrikNode.cpp

```diff
@@ -0,0 +1,660 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "BlendTreeFabrikNode.h"
+#include "AnimGraphManager.h"
+#include "EventManager.h"
+#include "Node.h"
+#include "Recorder.h"
+#include "TransformData.h"
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <EMotionFX/Source/AnimGraph.h>
+#include <EMotionFX/Source/DebugDraw.h>
+#include <MCore/Source/LogManager.h>
+#include <MCore/Source/Vector.h>
+
+namespace EMotionFX
+{
+    AZ_CLASS_ALLOCATOR_IMPL(BlendTreeFabrikNode, AnimGraphAllocator)
+    AZ_CLASS_ALLOCATOR_IMPL(BlendTreeFabrikNode::UniqueData, AnimGraphObjectUniqueDataAllocator)
+
+    BlendTreeFabrikNode::UniqueData::UniqueData(AnimGraphNode* node, AnimGraphInstance* animGraphInstance)
+        : AnimGraphNodeData(node, animGraphInstance)
+    {
+    }
+
+    void BlendTreeFabrikNode::UniqueData::Update()
+    {
+        BlendTreeFabrikNode* fabrikIKNode = azdynamic_cast<BlendTreeFabrikNode*>(m_object);
+        AZ_Assert(fabrikIKNode, "Unique data linked to incorrect node type.");
+
+        const ActorInstance* actorInstance = m_animGraphInstance->GetActorInstance();
+        const Actor* actor = actorInstance->GetActor();
+        const Skeleton* skeleton = actor->GetSkeleton();
```

Comment:
is it possible for any of these ever to be nullptr?
If not, great
If so, consider checking, and returning...

Reply:
All the existing blend tree nodes I looked at don't check for nullptr here, so I assume it will never be null.